### DISCUSSION TITLE
Template specialisation syntax

### DISCUSSION THREAD

**Discussion by tcbrindle** [hsutter/cppfront#1284] [Q&A] [answered]
What is the correct syntax for writing explicit and/or partial specialisations of class and variable templates in Cpp2?

(Apologies if this is covered in the docs somewhere, I looked but couldn't find it.)

**Comment by MaxSagebaum** [comment, marked as answer]
It is currently not possible. There is merge request which addresses this https://github.com/hsutter/cppfront/pull/598.

If this is merged the syntax would look like: https://cpp2.godbolt.org/z/KP4cGjscK
```
A: <E,C> type = {
    print: (this) = {
        std::cout << "A normal" << std::endl;
    } 
}

A: <C> specialize<C,C> type = {
    print: (this) = {
        std::cout << "A specialized" << std::endl;
    }
}

main: () = {
    a1: A<double, int> = ();
    a2: A<double, double> = ();

    a1.print();
    a2.print();
}
```

**Reply by tcbrindle** [reply]
Thanks for the info 👍
